### PR TITLE
[Homework #18] - React UI by Max Hirning

### PR DESCRIPTION
**Pull request by Max-Hirning**

### CODE REVIEW

File: homeworks/max-hirning/18-react-ui/src/views/components/AddButton.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    return {};
+  }
+
+  return (
+    <button 
+      onClick={addTile}
+      disabled={loading}
+      className={styles.button}
+      style={{...disabledStyle()}}
```
Comment (ilya-ischenko) [inline homeworks/max-hirning/18-react-ui/src/views/components/AddButton.tsx:29]: Can be replaced with ternary operator. loading ? { opacity: 0.5 } : {}.

File: homeworks/max-hirning/18-react-ui/src/views/components/Tile.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  const refreshTile = () => {
+    setShowRefreshBtn(true);
+    setIsRotating(true);
+    setTimeout(() => {
+      setIsRotating(false);
+      setShowRefreshBtn(false);
+      dispatch(refreshOne(id));
+    }, 1500);
+
+  }
```
Comment (ilya-ischenko) [inline homeworks/max-hirning/18-react-ui/src/views/components/Tile.tsx:29]: So this odd setTimeout can be removed after you change refresh functionality

**Review by ilya-ischenko — COMMENTED** [review]
1. Typo in file name - "tilse.ts". homeworks/max-hirning/18-react-ui/src/redux/tilse.ts
2. Keep attention to the styles. "Refresh all button" is on the very bottom of the page, there should be some space between it and boundary.
3. Refresh and refresh all buttons have wrong behaviour. They should update tile/s. When you click "refresh all" button it should update all tiles with new pictures. And when you click update button on a single tile it should update corresponding tile with a new image.

File: homeworks/max-hirning/18-react-ui/src/views/components/AddButton.tsx (hunk 1 above)
Comment (Max-Hirning) [reply homeworks/max-hirning/18-react-ui/src/views/components/AddButton.tsx]: Did with function, 'cauze always implement style object(in such situation function is more understandable as for me), but here did automatically.
Fixed.

File: homeworks/max-hirning/18-react-ui/src/views/components/Tile.tsx (hunk 2 above)
Comment (Max-Hirning) [reply homeworks/max-hirning/18-react-ui/src/views/components/Tile.tsx]: Fixed

**Comment by Max-Hirning** [comment]
> 1. Typo in file name - "tilse.ts". homeworks/max-hirning/18-react-ui/src/redux/tilse.ts
> 2. Keep attention to the styles. "Refresh all button" is on the very bottom of the page, there should be some space between it and boundary.
> 3. Refresh and refresh all buttons have wrong behaviour. They should update tile/s. When you click "refresh all" button it should update all tiles with new pictures. And when you click update button on a single tile it should update corresponding tile with a new image.

**Comment by Max-Hirning** [comment]
Fixed my issue with refreshing

**Review by ilya-ischenko — APPROVED** [review]
Well done!
Approved.

**Pull request merged**